### PR TITLE
pythonPackages.{sseclient-py,python-nest}: init

### PR DESCRIPTION
**Pull request by JamieMagee**
###### Motivation for this change

Support `nest` in home assistant

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-nest/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ buildPythonPackage, fetchFromGitHub, lib, python, python-dateutil, requests
+, six, sseclient-py }:
+
+buildPythonPackage rec {
+  pname = "python-nest";
+  version = "4.1.0";
+
+  # pypi does not contain tests, using github sources instead
```
Comment (mweinelt) [inline pkgs/development/python-modules/python-nest/default.nix:8]: This comment does not really make sense when there are no tests at all. In which case we can fetch this package from pypi.

File: pkgs/development/python-modules/python-nest/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "jkoelker";
+    repo = "python-nest";
+    rev = "b45100440577ca35061e08dda48fddcc49f214eb";
```
Comment (mweinelt) [inline pkgs/development/python-modules/python-nest/default.nix:12]: Let's use the tag here.

```suggestion
    rev = "v${version}";
```

File: pkgs/development/python-modules/python-nest/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "0w5iv08pvlb79q078lscd2wpy5i3m3hynl3lkw8pv5l2ik3ln3i1";
+  };
+
+  propagatedBuildInputs = [ python-dateutil requests six sseclient-py ];
+
+  # no tests
+  doCheck = false;
```
Comment (mweinelt) [inline pkgs/development/python-modules/python-nest/default.nix:19]: Tests are important to know early when something breaks. If there are no tests there should be at least a `pythonImportsCheck`.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck

**Review by mweinelt — CHANGES_REQUESTED** [review]
I added some remarks, remember that you need to update the home-assistant component packages file through `parse-requirement.py`. That usually happens in a separate commit, since it touches the home-assistant package.

**Comment by JamieMagee** [comment]
Thanks for all the assistance. I'm still getting the hang of creating packages, and I had been reusing my work from previous python packages. I should be in a better position for future packages.

File: pkgs/development/python-modules/python-nest/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  pname = "python-nest";
+  version = "4.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "12iyypbl92ybh8w1bf4z0c2g0sb9id2c07c89vzvnlxgjylw3wbi";
+  };
+
+  propagatedBuildInputs = [ python-dateutil requests six sseclient-py ];
+  pythonImportsCheck = [ "nest" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/python-nest/default.nix:14]:
```suggestion
  # has no tests
  doCheck = false;
  pythonImportsCheck = [ "nest" ];
```

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -5762,6 +5762,8 @@ in {
       inherit (pkgs) pkgconfig;
     };
 
+  python-nest = callPackage ../development/python-modules/python-nest { };
```
Comment (mweinelt) [inline pkgs/top-level/python-packages.nix:5765]: This should probably be above pythonnet, no?

Comment (JamieMagee) [reply pkgs/top-level/python-packages.nix:5765]: Right you are :)

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 98641` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python27Packages.python-nest</li>
    <li>python27Packages.sseclient-py</li>
    <li>python37Packages.python-nest</li>
    <li>python37Packages.sseclient-py</li>
    <li>python38Packages.python-nest</li>
    <li>python38Packages.sseclient-py</li>
  </ul>
</details>

**Review by mweinelt — APPROVED** [review]

**Pull request merged**